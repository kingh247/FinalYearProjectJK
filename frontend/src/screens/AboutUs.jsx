import React from 'react';
import featureImage from '../assets/logo.jpg'; // Adjust the path and file extension

const AboutUs = () => {
  return (
    <>
      <h1> About Us </h1>
      <div style={{ textAlign: 'center' }}>
        <img
          src={featureImage} 
          alt=""
          style={{ width: '100%', maxWidth: '730px', margin: '0 auto' }}
        />
      </div>
      <div>
        <p>
          Final year student trying to build a generic emcomerce webite that can
          be used for business to compete in the new every growing digital
          market place
        </p>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
          eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
          minim veniam, quis nostrud exercitation ullamco laboris nisi ut
          aliquip ex ea commodo consequat. Duis aute irure dolor in
          reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
          pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
          culpa qui officia deserunt mollit anim id est laborum.
        </p>
      </div>
    </>
  );
};

export default AboutUs;

