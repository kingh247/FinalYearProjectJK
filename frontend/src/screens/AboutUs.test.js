import React from 'react';
import { render, screen } from '@testing-library/react';
import AboutUs from './AboutUs';

describe('AboutUs', () => {
  test('renders About Us heading', () => {
    render(<AboutUs />);
    const heading = screen.getByRole('heading', { name: /about us/i });
    expect(heading).toBeInTheDocument();
  });
  test('renders Features component with Lorem Ipsum text', () => {
    render(<AboutUs />);

    expect(screen).toBeTruthy();
  });

  test('renders description paragraphs', () => {
    render(<AboutUs />);
    const paragraphs = screen.getAllByText(/lorem ipsum/i);
    expect(paragraphs.length).toBe(1);
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<AboutUs />);
    expect(asFragment()).toMatchSnapshot();
  });
});
