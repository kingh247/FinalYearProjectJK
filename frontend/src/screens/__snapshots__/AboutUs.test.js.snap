// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs matches snapshot 1`] = `
<DocumentFragment>
  <h1>
     About Us 
  </h1>
  <div
    style="text-align: center;"
  >
    <img
      alt=""
      src="logo.jpg"
      style="width: 100%; max-width: 300px; margin: 0px auto;"
    />
  </div>
  <div>
    <p>
      Final year student trying to build a generic emcomerce webite that can be used for business to compete in the new every growing digital market place
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
  </div>
</DocumentFragment>
`;
